packages/edit-post/src/components/welcome-guide-modal/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { __experimentalCreateInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Guide from './guide';
+import { CanvasImage, EditorImage, BlockLibraryImage, InserterIconImage } from './vectors';
+
+export default function WelcomeGuideModal() {
+	const areTipsEnabled = useSelect( ( select ) => select( 'core/nux' ).areTipsEnabled() );
+
+	const { disableTips } = useDispatch( 'core/nux' );
+
+	if ( ! areTipsEnabled ) {
+		return null;
+	}
+
+	return (
+		<Modal
+			className="edit-post-welcome-guide-modal"
+			shouldCloseOnClickOutside={ false }
+			onRequestClose={ disableTips }
+		>
+			<Guide onFinish={ disableTips }>
+				<Guide.Page>
+					<h1 className="edit-post-welcome-guide-modal__heading">
+						{ __( 'Welcome to the block editor' ) }
+					</h1>
+					<CanvasImage className="edit-post-welcome-guide-modal__image" />
+					<p className="edit-post-welcome-guide-modal__text">
+						{ __( 'In the WordPress editor, each paragraph, image, or video is presented as a distinct “block” of content.' ) }
+					</p>
+				</Guide.Page>
+				<Guide.Page>
+					<h1 className="edit-post-welcome-guide-modal__heading">
+						{ __( 'Make each block your own' ) }
+					</h1>
+					<EditorImage className="edit-post-welcome-guide-modal__image" />
+					<p className="edit-post-welcome-guide-modal__text">
+						{ __( 'Each block comes with its own set of controls for changing things like color, width, and alignment. These will show and hide automatically when you have a block selected.' ) }
+					</p>
+				</Guide.Page>
+				<Guide.Page>
+					<h1 className="edit-post-welcome-guide-modal__heading">
+						{ __( 'Get to know the block library' ) }
+					</h1>
+					<BlockLibraryImage className="edit-post-welcome-guide-modal__image" />
+					<p className="edit-post-welcome-guide-modal__text">
+						{ __experimentalCreateInterpolateElement(
+							__( 'All of the blocks available to you live in the Block Library. You’ll find it wherever you see the <InserterIconImage /> icon.' ),
+							{
+								InserterIconImage: (
```

Comment:
I like how you chose to name the property to align with the component name, but in-so-doing it occurs to me whether we ought to have any coding guidelines around the names we choose for these interpolated keys? cc @nerrad

Reply:
I think this is a question best answered by those working more closely with translators.  

One of the intents behind the interface is to provide additional guidance to translators on what a tag represents and signal that it shouldn't be translated.  With WordPress core in particular, I think generally translators know to leave tags alone (although that has been a smaller subset of tags I think).   

I think others might be better equipped to detail what standards we should have.  cc @ocean90 , @swissspidy  or @akirk ,  do you have any guidance here?

Reply:
I am huge proponent of "speaking" component placeholders. It's a very easy way to convey to translators what is being substituted here. `InserterIconImage` sounds good to me.

How and where would be put coding guidelines for this?

Reply:
I like this proposal. Assuming that it's copied to all translations it would work great 👍 

For the full picture, there are some drawbacks discussed in the PR which test this new API. The discussion triggered by @johnbillion starts with this comment: https://github.com/WordPress/gutenberg/pull/18623#issuecomment-555995121.

Reply:
@akirk do you suggest that the following:

```js
{ __experimentalCreateInterpolateElement(
	__( 'While writing, you can press <kbd>/</kbd> to quickly insert new blocks.' ),
	{ kbd: <kbd /> }
) }
```

could be easier refactored to:
```js
```js
{ __experimentalCreateInterpolateElement(
	__( 'While writing, you can press <Shortcut> to quickly insert new blocks.' ),
	{ Shortcut: <kbd>/</kbd> }
) }
```